[c10d] s/torch::autograd::variable/at::Tensor/g

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54764 [DDP] Mark a few variables as const in reducer
* **#54763 [c10d] s/torch::autograd::variable/at::Tensor/g**

Replaces deprecated torch::autograd::variable with at::Tensor.
torch::autograd::variable is defined as equal to at::Tensor now so this should
be a noop, but follows convention of using tensor instead of Variable.

There are also a few uses in distributed autograd c++, can remove those in a separate diff

Differential Revision: [D27356450](https://our.internmc.facebook.com/intern/diff/D27356450/)

rohan-varma committed:
[c10d] s/torch::autograd::variable/at::Tensor/g

Replaces deprecated torch::autograd::variable with at::Tensor.
torch::autograd::variable is defined as equal to at::Tensor now so this should
be a noop, but follows convention of using tensor instead of Variable.

Differential Revision: [D27356450](https://our.internmc.facebook.com/intern/diff/D27356450/)

[ghstack-poisoned]

wayi1 reviewed (approved):
Thanks for deprecating Variable in our codebase!

rohan-varma committed:
Update on "[c10d] s/torch::autograd::variable/at::Tensor/g"


Replaces deprecated torch::autograd::variable with at::Tensor.
torch::autograd::variable is defined as equal to at::Tensor now so this should
be a noop, but follows convention of using tensor instead of Variable.

There are also a few uses in distributed autograd c++, can remove those in a separate diff

Differential Revision: [D27356450](https://our.internmc.facebook.com/intern/diff/D27356450/)

[ghstack-poisoned]
